presto-parser/src/main/java/com/facebook/presto/sql/parser/AstBuilder.java

```diff
@@ -1218,6 +1218,12 @@ public Node visitSpecialDateTimeFunction(SqlBaseParser.SpecialDateTimeFunctionCo
         return new CurrentTime(getLocation(context), type);
     }
 
+    @Override
+    public Node visitCurrentUserFunction(SqlBaseParser.CurrentUserFunctionContext context)
+    {
+        return new FunctionCall(getLocation(context.CURRENT_USER()), QualifiedName.of("current_user"), ImmutableList.of());
```

Comment:
This is not ideal, since it's a syntactic structure. The conversion to a function call should be done during SQL->IR translation. However, this is how we currently do it in other places, so let's keep it as is.

Reply:
Is there an example of the better way to do this?